packages/lesswrong/components/comments/CommentsList.jsx

```diff
@@ -1,45 +1,81 @@
 import { Components, replaceComponent, withEdit } from 'meteor/vulcan:core';
-import React from 'react';
+import React, { PureComponent } from 'react';
 import { FormattedMessage } from 'meteor/vulcan:i18n';
 import { Comments } from "meteor/example-forum";
+import { shallowEqual, shallowEqualExcept } from '../../lib/modules/utils/componentUtils';
 
-const CommentsList = ({
-  comments,
-  currentUser,
-  highlightDate,
-  editMutation,
-  post,
-  frontPage,
-}) => {
-  if (comments) {
-    return (
-      <Components.ErrorBoundary>
+class CommentsList extends PureComponent {
+  constructor(props, context) {
+    super(props)
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if(!shallowEqual(this.state, nextState))
+      return true;
+    if(!shallowEqualExcept(this.props, nextProps, ["post","comments","editMutation"]))
```

Comment:
I expect a bug here, where when you switch between two posts without comments, the CommentsList doesn't pass the new post argument through to its children, which might cause some weird bugs in the display, though I guess it might not turn out to be a problem in practice.

Reply:
Hmm, maybe. It would definitely be a problem if `post` were excluded from comparison in `CommentsListSection`, but `CommentsList` is probably fine. Added a comparison of `post._id` as a defensive measure.